dashboard/src/components/DeploymentForm/DeploymentForm.tsx

```diff
@@ -75,19 +72,27 @@ class DeploymentForm extends React.Component<IDeploymentFormProps, IDeploymentFo
   }
 
   public componentWillReceiveProps(nextProps: IDeploymentFormProps) {
-    const { selectChartVersionAndGetFiles, chartVersion } = this.props;
-    const { versions } = this.props.selected;
-    const { values } = nextProps.selected;
+    const { chartID, chartVersion, getChartValues, getChartVersion, hr, selected } = this.props;
+    const { version } = selected;
 
-    if (nextProps.chartVersion !== chartVersion) {
-      const cv = versions.find(v => v.attributes.version === nextProps.chartVersion);
-      if (cv) {
-        selectChartVersionAndGetFiles(cv);
-      } else {
-        throw new Error("could not find chart");
+    if (chartVersion !== nextProps.chartVersion) {
```

Comment:
this will only get triggered when switching the version in a new deployment (props.chartVersion changes)